src/plugins/dna_export/src/ExportSequenceTask.cpp

```diff
@@ -109,29 +109,38 @@ void ExportSequenceItem::releaseOwnedSeq() {
     }
 }
 
+bool ExportSequenceItem::isRefAlreadyCounted(const U2EntityRef& seqRef) {
```

Comment:
How this method is different from 'sequencesRefCount.contains(seqRef)' ?

Reply:
My method use operator == which work correct.
QMap::contains() use hash method or something like what  leads to wrong result - it returns 'true' for object which is not present in container.

Reply:
sequencesRefCounts is a QMap. 
Is it safe to have 'U2EntityRef' as a key in QMap? 

If yes: your code is not needed and map.contains(key) should be enough.

If it is not safe: this is the real cause of the bug: we need to make U2EntityRef work with QMap as expected or never use it (across all our code) as a key in QMap.